sdcm/utils/quota.py

```diff
@@ -0,0 +1,113 @@
+        return False
+    return True
+
+
+def get_currently_used_space_by_scylla_user(node) -> int:
+    result = node.remoter.sudo("xfs_quota -x -c 'report' /var/lib/scylla | grep scylla | tail -1")
+    used_space = int(result.stdout.split()[1])
+    return used_space
+
+
+def configure_quota_on_node_for_scylla_user(node) -> int:
+    """Configure quota with current disk usage size by scylla user + 1GB on /var/lib/scylla mount point"""
+    used_space = get_currently_used_space_by_scylla_user(node)
+    LOGGER.debug('Currently used space by scylla user is: {}K'.format(used_space))
+    quota_size = int(used_space + 1000000)
+    LOGGER.info(
+        "Configuring quota for scylla user with quota size: {}K on node {}".format(quota_size, node))
+    conf_quota_cmd = f"xfs_quota -x -c 'limit bsoft={quota_size}K bhard={quota_size}K scylla' /var/lib/scylla"
+    node.remoter.sudo(conf_quota_cmd, verbose=True, )
+    return quota_size
+
+
+def verify_quota_is_configured_for_scylla_user(node):
+    """Verifies quota ia actually configured for scylla user, but doesn't verify enforcement in ON"""
+    get_scylla_user_quota_value = 'xfs_quota -x -c "report -h" /var/lib/scylla | grep scylla | tail -1 | awk {\'print $4\'}'
+    LOGGER.info("Verifying quota is configured for scylla user")
+    quota_value = node.remoter.sudo(get_scylla_user_quota_value).stdout.rstrip()
+    if quota_value == "0":
+        LOGGER.error("Failed to configure quota on user")
+        raise QuotaConfigurationFailure(
+            "Quota was not configured for scylla user. The value for hard quota is: {}".format(quota_value))
+    LOGGER.info("Quota was configured for scylla user with value: {}".format(quota_value))
+
+
+def remove_quota_configuration_from_node(node):
+    LOGGER.info('Remove quota configuration from user \'scylla\' by setting it to 0')
+    quota_remove_cmd = "xfs_quota -x -c 'limit bsoft=0 bhard=0 scylla' /var/lib/scylla"
+    node.remoter.sudo(quota_remove_cmd, verbose=True)
+
+
+def write_data_to_reach_end_of_quota(node, quota_size):
+    LOGGER.debug('Watching the log for the relevant message for quota...')
+    quota_exceeded_appearances = node.follow_system_log(
+        patterns=[r'Shutting down communications due to I\/O errors until operator intervention: Disk error:.*Disk quota exceeded'])
+    node.remoter.sudo('chsh -s /bin/bash scylla')  # Enable login for scylla user
+
+    def approach_end_of_quota():
+        if bool(list(quota_exceeded_appearances)):
+            return True
+        currently_used_space = get_currently_used_space_by_scylla_user(node)
+        occupy_space_size = int((quota_size - currently_used_space) * 90 / 100)
+        occupy_space_cmd = f"su - scylla -c 'fallocate -l {occupy_space_size}K /var/lib/scylla/occupy_90percent.{time.time()}'"
+        with EventsSeverityChangerFilter(new_severity=Severity.NORMAL,  # End of quota created CRITICAL event
+                                         event_class=CassandraStressEvent,
```

Comment:
how about other load tools like s-b?
Also, shouln't be done as a part of the `ignore_disk_quota_exceeded_errors` so in case c-s fails in between we don't create critical?

Reply:
why would c-s gonna fail ? or have critical events in this case ? i.e. its one node going down.

c-s need to continue without any issues, if not, we should open a bug, I don't think we do this during regular ENSPC nemesis